wear/src/main/java/io/homeassistant/companion/android/home/views/SensorUi.kt

```diff
@@ -21,13 +24,43 @@ fun SensorUi(
     basicSensor: SensorManager.BasicSensor,
     onSensorClicked: (String, Boolean) -> Unit,
 ) {
+    var checked = sensor?.enabled == true
+
+    val backgroundRequest =
+        rememberLauncherForActivityResult(ActivityResultContracts.RequestPermission()) {}
+
+    val permissionLaunch = rememberLauncherForActivityResult(
+        ActivityResultContracts.RequestMultiplePermissions()
+    ) { isGranted ->
+        var allGranted = true
+        isGranted.forEach {
+            if (manager.requiredPermissions(basicSensor.id)
+                .contains(Manifest.permission.ACCESS_FINE_LOCATION) && manager.requiredPermissions(basicSensor.id)
+                    .contains(Manifest.permission.ACCESS_BACKGROUND_LOCATION) && it.key == Manifest.permission.ACCESS_FINE_LOCATION
+            )
+                backgroundRequest.launch(Manifest.permission.ACCESS_BACKGROUND_LOCATION)
```

Comment:
As this permission only exists on API level 29 or higher, a version check should be added [like in the main app](https://github.com/home-assistant/android/blob/93844ffea03be01e35c618114073b0e26fd47f39/app/src/main/java/io/homeassistant/companion/android/settings/sensor/SensorDetailViewModel.kt#L345-L350).